src/javatests/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/ExtractExecutorImplTest.java

```diff
@@ -247,6 +249,95 @@ public void run_incrementalMode_success() throws Exception {
             eq(5000),
             eq(5 + 1));
     verifyNoMoreInteractions(scriptManager);
+    verify(saveChecker).getScriptCheckPoints(Paths.get("test_path"));
+    verifyNoMoreInteractions(saveChecker);
+  }
+
+  @Test
+  public void run_recoveryModeNoChunk_success() throws Exception {
+    ImmutableSet<String> targetScripts = ImmutableSet.of("script_no_record", "script_with_record");
+    when(scriptManager.getAllScriptNames()).thenReturn(targetScripts);
+    when(schemaManager.getSchemaKeys(any(Connection.class), eq(ImmutableList.of())))
+        .thenReturn(ImmutableSet.of());
+    when(saveChecker.findScriptNamesWithFinishedRecords(
+            any(Path.class), eq(targetScripts), any(String.class)))
+        .thenReturn(ImmutableSet.of("script_with_record"));
+    Arguments arguments =
+        Arguments.builder()
+            .setDbConnectionProperties(properties)
+            .setDbConnectionAddress("jdbc:hsqldb:mem:recovery-mode-no-chunk.example")
+            .setOutputPath(Paths.get("/tmp"))
+            .setMode(RunMode.RECOVERY)
+            .setChunkRows(0)
+            .setPrevRunPath(Paths.get("test_path"))
+            .build();
+
+    assertThat(executor.run(arguments)).isEqualTo(0);
+
+    verify(scriptManager).getAllScriptNames();
+    verify(scriptManager)
+        .executeScript(
+            any(Connection.class),
+            /*dryRun=*/ eq(false),
+            any(SqlTemplateRenderer.class),
+            /*scriptName=*/ eq("script_no_record"),
+            eq(dataEntityManager),
+            eq(0),
+            eq(0));
+    verifyNoMoreInteractions(scriptManager);
```

Comment:
nit: such an assertion may have less sense if do not expect calls to other methods of the class – e.g.:

```
bool fooBar = scriptManager.foo();
if (fooBar) {
  scriptManager.bar();
}
```

Otherwise, single `verify` statement is enough and `verifyNoMoreInteractions` looks slightly over-defensive

Reply:
Actually for the "no chunk" test we do want to verify the chunk-related method is not called (i.e. RECOVERY mode is decoupled from chunking requirements). For the "chunk" test it is indeed true and I removed verifyNoMoreInteractions checks.